client/me/purchases/remove-purchase/index.jsx

```diff
@@ -117,32 +228,100 @@ const RemovePurchase = React.createClass( {
 				className="remove-purchase__dialog"
 				isVisible={ this.state.isDialogVisible }
 				onClose={ this.closeDialog }>
-				<h1>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</h1>
-				{ this.renderDialogText() }
+				<FormSectionHeading>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</FormSectionHeading>
+				{ this.renderDomainDialogText() }
 			</Dialog>
 		);
 	},
 
-	renderDialogText() {
+	renderDomainDialogText() {
 		const purchase = getPurchase( this.props ),
 			productName = getName( purchase );
 
-		if ( isDomainRegistration( purchase ) ) {
-			return (
-				<p>
-					{
-						this.translate( 'This will remove %(domain)s from your account.', {
-							args: { domain: productName }
-						} )
-					}
-					{ ' ' }
-					{ this.translate( 'By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.' ) }
-				</p>
-			);
+		return (
+			<p>
+				{
+					this.translate( 'This will remove %(domain)s from your account.', {
+						args: { domain: productName }
+					} )
+				}
+				{ ' ' }
+				{ this.translate( 'By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.' ) }
+			</p>
+		);
+	},
+
+	renderPlanDialogs() {
+		const buttons1 = [
+			{
+				action: 'cancel',
+				disabled: this.state.isRemoving,
+				label: this.translate( 'Cancel' )
+			},
+			{
+				action: 'next-2',
+				disabled: this.state.isRemoving || this.state.questionOneRadio == null || this.state.questionTwoRadio == null,
+				label: this.translate( 'Next' ),
+				onClick: this.displaySurveyTwo
+			} ],
+			buttons2 = [
+				{
+					action: 'cancel',
+					disabled: this.state.isRemoving,
+					label: this.translate( 'Cancel' )
+				},
+				{
+					action: 'prev-1',
+					disabled: this.state.isRemoving,
+					label: this.translate( 'Previous' ),
+					onClick: this.displaySurveyOne
+				},
+				{
+					action: 'remove',
+					disabled: this.state.isRemoving,
+					isPrimary: true,
+					label: this.translate( 'Remove' ),
+					onClick: this.removePurchase
+				} ],
+			productName = getName( getPurchase( this.props ) );
+
+		if ( ! config.isEnabled( 'upgrades/removal-survey' ) ) {
+			buttons2.splice( 1, 1 );
 		}
 
-		let includedDomainText;
+		return (
+			<div>
+				<Dialog
+					buttons={ buttons1 }
+					className="remove-purchase__dialog"
+					isVisible={ this.state.isDialogVisible && this.state.isSurveyOneVisible }
```

Comment:
They don't directly control visibility of the surveys. I understand you're using these values in combination with `isDialogVisible` to determine visibility. But it makes their names inaccurate, because you can have `isSurveyOneVisible = true` while survey one is not actually visible. It's simple enough to understand when looking at this line, but it's bound to be confusing when someone is just looking at the state values on their own (e.g., using the React inspector tools).

I think the structure in general is a little confusing, because we're trying to include the original survey as part of the one/two step process. This means when the config is false, we start off by showing dialog two which seems confusing to me.

I think it would be a little more clear if we separate the concept of which step you are on from the concept of whether or not the survey is enabled. We can track the survey step with a single variable, rather than two booleans. And it can always start at `1`. like `state.surveyStep = 1`.

Something like this:
```js
renderPlanDialogs() {
  const buttons = {
    cancel: {
      action: 'cancel',
      disabled: // blah blah blah
    },
    next: {
      action: 'next-2',
      // blah blah blah
    },
    previous: { /* blah blah blah */ },
    remove: { /* blah blah blah */ },
  }
  const inStepOne = state.surveyStep === 1;

  if ( ! config.isEnabled( 'upgrades/removal-survey' ) ) {
    const buttonsArr = [ buttons.cancel, buttons.remove ];
    const dialogContent = this.renderPlanDialogsText();
  } else {
    buttonsArr = ( inStepOne ) ? [ buttons.cancel, buttons.next, buttons.remove ] : [ buttons.cancel, buttons.previous, buttons.remove ];
    dialogContent = ( inStepOne ) ? this.renderDialogContentOne() : this.renderDialogContentTwo();
  }

  return ( 
    <Dialog
      buttons={ buttonsArr }
      className="remove-purchase__dialog"
      isVisible={ this.state.isDialogVisible }
      onClose={ this.closeDialog }
      transitionEnter={ false }
      transitionLeave={ false }>
      <FormSectionHeading>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</FormSectionHeading>
      { dialogContent }
    </Dialog> );
}

      
```